feat(http): Support for per-host http concurrency limits (#6983)
renovatebot/renovate: docs/usage/configuration-options.md

If you have any uncertainty about exactly which hosts a service uses, then it can be more reliable to use `domainName` instead of `hostName` or `baseUrl`. e.g. configure `"hostName": "docker.io"` to cover both `index.docker.io` and `auth.docker.io` and any other host that's in use.

### concurrentRequestLimit

> Limit number of concurrent requests on per-host basis.

Author: Fixme

Renovate does not do a "longest match" algorithm to pick between multiple matching `baseUrl` values in different rules, so put the longer `baseUrl` rule _after_ the shorter one in your `hostRules`.

### concurrentRequestLimit

Usually the default setting is fine, but you can use `concurrentRequestLimit` to limit the number of concurrent outstanding requests.
You only need to adjust this setting if a datasource is rate limiting Renovate or has problems with the load.
The limit will be set for any host it applies to.

> Example:

Reviewer: ```suggestion
e.g.:
```

@rarkins do you like `Example:` or `e.g.:` better here?

Reviewer 2: I think maybe "Example config:" is more verbose and readable

Reviewer: ```suggestion
Example config:
```

Let's go with this then. 😄

Renovate does not do a "longest match" algorithm to pick between multiple matching `baseUrl` values in different rules, so put the longer `baseUrl` rule _after_ the shorter one in your `hostRules`.

### concurrentRequestLimit

> Usually the default setting is fine, but you can use `concurrentRequestLimit` to limit the number of concurrent
> outstanding requests.

Reviewer: ```suggestion
Usually the default setting is fine, but you can use `concurrentRequestLimit` to limit the number of concurrent outstanding requests.
```

Fix so that we conform to the "one sentence per line" system.

Author: Ouch, that's my code reformatting

If you have any uncertainty about exactly which hosts a service uses, then it can be more reliable to use `domainName` instead of `hostName` or `baseUrl`.
e.g. configure `"hostName": "docker.io"` to cover both `index.docker.io` and `auth.docker.io` and any other host that's in use.

### concurrentRequestLimit

> Limit number of concurrent requests on per-host basis.

Reviewer 2: ```suggestion
Setting this value will limit the number of concurrent outstanding requests for any host it applies to. It should be used only if a datasource is rate limiting Renovate or having problems with the load.
```

> Setting this value will limit the number of concurrent outstanding requests for any host it applies to. It should be used only if a datasource is rate limiting Renovate or having problems with the load.

Reviewer 3: @HonkingGoose Can you check this please.

> Setting this value will limit the number of concurrent outstanding requests for any host it applies to. It should be used only if a datasource is rate limiting Renovate or having problems with the load.

Reviewer: ```suggestion
Usually the default `concurrentRequestLimit` is fine, but you can use `concurrentRequestLimit` to limit the number of concurrent outstanding requests.
You only need to adjust this setting if a datasource is rate limiting Renovate or has problems with the load.
The limit will be set for any host it applies to.
```

How about something like this?

Author: @HonkingGoose I like this forumlation, also I'll add example snippet
